Mlem/API/Models/ListingType.swift

```diff
@@ -11,42 +11,5 @@ enum APIListingType: String, Codable {
     case all = "All"
     case local = "Local"
     case subscribed = "Subscribed"
-    
-    // Pre 0.18.0 it appears that they used integers instead of strings here. We can remove this intialiser once we drop support for old versions. To fully support both systems, we'd also need to *encode* back into the correct integer or string format. I'd rather not go through the effort for instance versions that most people don't use any more, so I've disabled the option to edit account settings on instances running <0.18.0
-    // - sjmarf
-    
-    // TODO: 0.17 deprecation remove this initialiser
-    
-    init(from decoder: Decoder) throws {
```

Comment:
We can't remove this decoder just yet. 

Inside of `MyUserView` we have to decode a `ListingType` in inteher form on pre-0.18.0 versions. This decoder handles that. We can't just exclude the property entirely, unfortunately - due to a bug in 0.18.0 we need to send the `ListingType` when we save the user's settings - even if they aren't modifying that setting. This bug is fixed in 0.19.0.

If we remove this a decoding error is thrown and the feed won't load on test-mlem.jo.wtf (which runs 0.17.3)